### PR TITLE
A new package: cudnn.

### PR DESCRIPTION
**Pull request by weijianwen**
In #3235, @muffgaga hasn't replied on the refinement proposals for about 3 days. It is quite urgent for me to use cudnn thus I resubmit a PR with slight modification. 

**LIMITATION**: As the file name `cudnn-8.0-linux-x64-v5.1.tgz` indicates, this package works for `cudnn@5.1` against `cuda@8.0` on x86_64 Linux platform only. Future work can be done to adapt it to more platforms.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cudnn/package.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class Cudnn(Package):
+    """NVIDIA cuDNN is a GPU-accelerated library of primitives for deep
+    neural networks"""
+
+    homepage = "https://developer.nvidia.com/cudnn"
+
+    version('5.1', '406f4ac7f7ee8aa9e41304c143461a69',
+            url='http://developer.download.nvidia.com/compute/redist/cudnn/'\
+            'v5.1/cudnn-8.0-linux-x64-v5.1.tgz')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cudnn/package.py]: You don't need to break URLs. Spack ignores URLs when running its line-length checks:
https://github.com/LLNL/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L73

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cudnn/package.py]: It looks like this line break is causing the flake8 tests to fail. Once you put everything on a single line, this package should be good to go.

**Comment by weijianwen** [comment]
Done. Latest version 6.0 has been added.

**Comment by weijianwen** [comment]
Done. Latest version 6.0 has been added.

File: var/spack/repos/builtin/packages/cudnn/package.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class Cudnn(Package):
+    """NVIDIA cuDNN is a GPU-accelerated library of primitives for deep
+    neural networks"""
+
+    homepage = "https://developer.nvidia.com/cudnn"
+
+    version('5.1', '406f4ac7f7ee8aa9e41304c143461a69',
+            url='http://developer.download.nvidia.com/compute/redist/cudnn/v5.1/cudnn-8.0-linux-x64-v5.1.tgz')
+    version('6.0', '4aacb7acb93c5e4dfa9db814df496219',
+            url='http://developer.download.nvidia.com/compute/redist/cudnn/v6.0/cudnn-8.0-linux-x64-v6.0.tgz')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cudnn/package.py]: Can you move 6.0 above 5.1 so things are sorted from newest to oldest?

Comment (weijianwen) [inline var/spack/repos/builtin/packages/cudnn/package.py]: Done

**Pull request merged**